lib/Cmts.ml

```diff
@@ -533,34 +533,39 @@ module Unwrapped = struct
     in
     vbox 0 ~name:"multiline" (list_fl unindented fmt_line $ fmt_opt epi)
 
-  let fmt ~ocp_indent_compat Cmt.{txt= s; loc} (pos : Cmt.pos) =
+  let fmt Cmt.{txt= s; loc} =
     let open Fmt in
     let is_sp = function ' ' | '\t' -> true | _ -> false in
     match String.split_lines (String.rstrip s) with
     | first_line :: (_ :: _ as tl) when not (String.is_empty first_line) ->
-        if ocp_indent_compat then
-          (* Not adding artificial breaks and keeping the comment contents
-             verbatim will not interfere with ocp-indent. *)
-          match pos with
-          | Before -> wrap "(*" "*)" @@ str s
-          | Within -> wrap "(*" "*)" @@ str s
-          | After -> break_unless_newline 1000 0 $ wrap "(*" "*)" @@ str s
-        else
-          let epi =
-            (* Preserve position of closing but strip empty lines at the
-               end *)
-            match String.rfindi s ~f:(fun _ c -> not (is_sp c)) with
-            | Some i when Char.( = ) s.[i] '\n' ->
-                break 1000 (-2) (* Break before closing *)
-            | Some i when i < String.length s - 1 ->
-                str " " (* Preserve a space at the end *)
-            | _ -> noop
-          in
-          (* Preserve the first level of indentation *)
-          let starts_with_sp = is_sp first_line.[0] in
-          wrap "(*" "*)"
-          @@ fmt_multiline_cmt ~opn_pos:loc.loc_start ~epi ~starts_with_sp
-               first_line tl
+        let epi =
+          (* Preserve position of closing but strip empty lines at the end *)
+          match String.rfindi s ~f:(fun _ c -> not (is_sp c)) with
+          | Some i when Char.( = ) s.[i] '\n' ->
+              break 1000 (-2) (* Break before closing *)
+          | Some i when i < String.length s - 1 ->
+              str " " (* Preserve a space at the end *)
+          | _ -> noop
+        in
+        (* Preserve the first level of indentation *)
+        let starts_with_sp = is_sp first_line.[0] in
+        wrap "(*" "*)"
+        @@ fmt_multiline_cmt ~opn_pos:loc.loc_start ~epi ~starts_with_sp
+             first_line tl
+    | _ -> wrap "(*" "*)" @@ str s
+end
+
+module Verbatim = struct
+  let fmt s (pos : Cmt.pos) =
+    let open Fmt in
+    match String.split_lines (String.rstrip s) with
+    | first_line :: _ :: _ when not (String.is_empty first_line) -> (
+      (* Not adding artificial breaks and keeping the comment contents
+         verbatim will not interfere with ocp-indent. *)
+      match pos with
+      | Before -> wrap "(*" "*)" @@ str s
+      | Within -> wrap "(*" "*)" @@ str s
+      | After -> break_unless_newline 1000 0 $ wrap "(*" "*)" @@ str s )
     | _ -> wrap "(*" "*)" @@ str s
```

Comment:
This can be simplified a lot: `wrap` and `str` are common to every cases, the only special case would be `fmt_if (pos = After && String.contains s '\n' && not (String.starts_with ~prefix:"\n" s)) (break_unless_newline 1000 0)`. I'm not even sure the newline checks are necessary.

Reply:
done